hummingbird/ml/operator_converters/skl_sv.py

```diff
@@ -0,0 +1,117 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import torch
+import scipy
+import numpy as np
+
+from onnxconverter_common.registration import register_converter
+
+"""
+Converters for scikit-learn linear models: SVC, NuSVC.  (LinearSVC is covered by linear_classifier.py)
+"""
+
+
+class SVC(torch.nn.Module):
+    def __init__(self, kernel, degree, sv, nv, a, b, gamma, coef0, classes, device):
+        super(SVC, self).__init__()
+        self.kernel = kernel
+        self.degree = degree
+        self.gamma = gamma
+        self.regression = False
+        sv = sv.toarray() if type(sv) == scipy.sparse.csr.csr_matrix else sv
+
+        self.sv = torch.nn.Parameter(torch.from_numpy(sv).double(), requires_grad=False)
+        self.sv_t = torch.nn.Parameter(torch.transpose(self.sv, 0, 1), requires_grad=False)
+        self.sv_norm = torch.nn.Parameter(-self.gamma * (self.sv ** 2).sum(1).view(1, -1), requires_grad=False)
+        self.coef0 = coef0
+        self.n_features = sv.shape[1]
+        self.a = a
+        self.b = torch.nn.Parameter(torch.nn.Parameter(torch.from_numpy(b.reshape(1, -1)).double()), requires_grad=False)
+        self.start = [sum(nv[:i]) for i in range(len(nv))]
+        self.end = [self.start[i] + nv[i] for i in range(len(nv))]
+        self.len_nv = len(nv)
+        true_classes, false_classes = zip(*[(i, j) for i in range(self.len_nv) for j in range(i + 1, self.len_nv)])
+        self.true_classes = torch.nn.Parameter(torch.IntTensor([true_classes]), requires_grad=False)
+        self.false_classes = torch.nn.Parameter(torch.IntTensor([false_classes]), requires_grad=False)
+        self.classes = torch.nn.Parameter(torch.IntTensor(classes), requires_grad=False)
+        self.perform_class_select = False
+        if min(classes) != 0 or max(classes) != len(classes) - 1:
+            self.perform_class_select = True
+        self.n_classes = len(classes)
+
+    def forward(self, x):
+        x = x.double()
+
+        if self.kernel == "linear":
+            k = torch.mm(x, self.sv_t)
+        elif self.kernel == "rbf":
+            # using quadratic expansion--susseptible to rounding-off errors
```

Comment:
Can you please add a reference to [this](http://www.robots.ox.ac.uk/~albanie/notes/Euclidean_distance_trick.pdf) paper in the comment?

Reply:
yupper.